app/uk/gov/hmrc/excisemovementcontrolsystemapi/controllers/DraftExciseMovementController.scala

```diff
@@ -65,7 +65,7 @@ class DraftExciseMovementController @Inject()(
       case _ => throw new Exception("invalid message sent to draft excise movement controller")
     }
 
-    workItemService.createWorkItem(newMovement.consignorId).flatMap { _ =>
+    workItemService.addWorkItemForErn(newMovement.consignorId).flatMap { _ =>
```

Comment:
Here if the workItem push fails, the movement is not saved but we may want to save it as the Message went through (EIS submission was successful). So here again raise the question that we may want to do a pushNew workitem also when the client ask for all message (GetALMewssages) if there are no in the cache. If this is the case then these could be done in parallel.